webcomponents-loader.js

```diff
@@ -31,44 +121,25 @@
   if (!('content' in document.createElement('template')) || !window.Promise || !Array.from ||
     // Edge has broken fragment cloning which means you cannot clone template.content
     !(document.createDocumentFragment().cloneNode() instanceof DocumentFragment)) {
-    polyfills = ['lite'];
+    polyfills = ['sd-ce-pf'];
   }
 
   if (polyfills.length) {
     var script = document.querySelector('script[src*="' + name +'"]');
     var newScript = document.createElement('script');
+    newScript.setAttribute('async', '');
     // Load it from the right place.
-    var replacement = 'webcomponents-' + polyfills.join('-') + '.js';
+    var replacement = 'bundles/webcomponents-' + polyfills.join('-') + '.js';
     var url = script.src.replace(name, replacement);
     newScript.src = url;
-    // NOTE: this is required to ensure the polyfills are loaded before
-    // *native* html imports load on older Chrome versions. This *is* CSP
-    // compliant since CSP rules must have allowed this script to run.
-    // In all other cases, this can be async.
-    if (document.readyState === 'loading' && ('import' in document.createElement('link'))) {
-      document.write(newScript.outerHTML);
-    } else {
-      document.head.appendChild(newScript);
-    }
+    newScript.addEventListener('load', function() {
+      ready();
```

Comment:
It seems like an unnecessary change from the previous behavior to divorce the `WebComponentsReady` event from `whenLoaded`. Specifically, previously `WebComponentsReady` used to mean "all imports that likely include the custom elements" have loaded. Now it just means "the polyfills have loaded and maybe some other stuff has."

Another related issue, it seems like if you never call `whenLoaded`, custom elements will never upgrade.

Perhaps instead we can:

1. fire the `WebComponentsReady` event after the Promise returned from `whenLoaded` resolves one is returned or when it is called. 

2. If `whenLoaded` has not been called by the time the document has loaded, call it.

This or some flavor of it would solve both issues.

Reply:
`1` is already what is happening.

Custom Elements will absolutely upgrade without calling `whenLoaded`. The only way they would never upgrade is if `whenLoaded` is called, and the function or promise given there never resolves.

Reply:
Or just remove the event if we can.